### DISCUSSION TITLE
k3s  registries.yaml rewrite usage

### DISCUSSION THREAD

**Discussion by codering** [k3s-io/k3s#10447] [Q&A]
k3s version
```
k3s version v1.29.6+k3s1 (83ae095a)
go version go1.21.11
```

My registries.yaml
```yaml
mirrors:
  "docker.io":
    endpoint:
      - "https://swr.cn-east-3.myhuaweicloud.com"
    rewrite:
      "(.*)": "hmirror/$1"
configs:
  "swr.cn-east-3.myhuaweicloud.com":
    auth:
      username: xxx
      password: yyy
```
I have a image `swr.cn-east-3.myhuaweicloud.com/hmirror/weaveworks/wego-app:v0.38.0 ` and I can pull it successfully.

```
crictl pull swr.cn-east-3.myhuaweicloud.com/hmirror/weaveworks/wego-app:v0.38.0
Image is up to date for sha256:44bfc91914c1b76ce4225dfea146957795533b4ffab6d838df8cfe087edcfda1
```

but when I pull image like this,  it failed.
```sh
crictl pull weaveworks/wego-app:v0.38.0
```
 error message is
```
E0704 14:18:00.983327    1458 remote_image.go:180] "PullImage from image service failed" 
err="rpc error: code = Unknown desc = failed to pull and unpack image \"docker.io/weaveworks/wego-app:v0.38.0\": 
failed to resolve reference \"docker.io/weaveworks/wego-app:v0.38.0\": failed to authorize: failed to fetch oauth token: 
unexpected status from GET request to https://swr.cn-east-3.myhuaweicloud.com/swr/auth/v2/registry/auth/?scope=repository%3Ahmirror%2Fweaveworks%2Fwego-app%3A&scope=repository%3Aweaveworks%2Fwego-app%3Apull&service=dockyard: 404 Not Found" 
image="weaveworks/wego-app:v0.38.0"
```

according to the [rewrite doc](https://docs.k3s.io/installation/private-registry#rewrites)
`docker.io/weaveworks/wego-app:v0.38.0` -> `swr.cn-east-3.myhuaweicloud.com/hmirror/weaveworks/wego-app:v0.38.0`

Where did I go wrong?

**Comment by brandond** [comment]
I don't know. Looks like you got a 404 when trying to auth to your mirror:
`failed to authorize: failed to fetch oauth token: 
unexpected status from GET request to https://swr.cn-east-3.myhuaweicloud.com/swr/auth/v2/registry/auth/?scope=repository%3Ahmirror%2Fweaveworks%2Fwego-app%3A&scope=repository%3Aweaveworks%2Fwego-app%3Apull&service=dockyard: 404 Not Found"`

I would probably take that up with whoever operates your mirror.
